src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -973,23 +978,17 @@ trait Scanners extends ScannersCommon {
     /** Convert current strVal, base to long value.
      *  This is tricky because of max negative value.
      *
-     *  Conversions in base 10 and 16 are supported. As a permanent migration
-     *  path, attempts to write base 8 literals except `0` emit a verbose error.
+     *  Conversions in base 2, 10 and 16 are supported.
      */
     def intVal(negated: Boolean): Long = {
       def intConvert: Long = {
         def malformed: Long = { syntaxError("malformed integer number") ; 0 }
         def tooBig: Long = { syntaxError("integer number too large") ; 0 }
-        val len = strVal.length
-        if (len == 0) {
-          if (base != 8) syntaxError("missing integer number")  // e.g., 0x;
-          0                                                     // 0 still looks like octal prefix
-        } else {
-          if (base == 8) {
-            if (settings.warnOctalLiteral)
-              deprecationWarning("Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)" , since="2.10")
-            base = 10
-          }
+        strVal.length match {
+        case 0 => syntaxError("missing integer number") ; 0L  // e.g., 0x;
+        case len =>
+          if (settings.warnOctalLiteral && strVal.charAt(0) == '0')
+            deprecationWarning("Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)", since="2.10")
```

Comment:
there's some argument that we ought to forbid integer literals with leading zeros, but this PR doesn't change anything in that regard

Reply:
It was put under a flag because someone preferred to allow, yea, even encourage them.

Reply:
Apparently I did it.